openquake/hazardlib/calc/filters.py

```diff
@@ -123,9 +123,17 @@ def get_distances(rupture, sites, param, dcache=None):
         dist = rupture.surface.get_azimuth(sites)
     elif param == 'azimuth_cp':
         dist = rupture.surface.get_azimuth_of_closest_point(sites)
-    elif param == 'closest_point':
-        t = rupture.surface.get_closest_points(sites)
-        dist = numpy.vstack([t.lons, t.lats, t.depths]).T  # shape (N, 3)
+    elif param == 'closest_point' or param == 'clon' or param == 'clat':
+        t = rupture.surface.get_closest_points(sites)  # TODO: measure the
+        # perfomance penalty due to the triple call to `.get_closest_points`
+        # In large calculations; it could very well be insignificant and in
+        # that case it is not worth doing anything
+        if param == 'closest_point':
+            dist = numpy.vstack([t.lons, t.lats, t.depths]).T  # shape (N, 3)
+        if param == 'clon':
+            dist = numpy.vstack([t.lons])  # shape (N, 3)
```

Comment:
The comment seems wrong, the shape cannot be (N, 3), perhaps (N, 1)? In that case it is much cleaner to use a .reshape than a .vstack. Same comment applies below to "clat".